docs/guide/areas-and-widgets/custom-widgets.md

```diff
@@ -1,116 +1,206 @@
 
-```javascript
-// app.js
-// After the other modules, configure a new one
-    'two-column-widget': {}
-```
+We will use the example of a two column **"layout widget."** It is a fairly common and relatively simple use case that allows editors to visually align content in a row. This version of a layout widget consists of two areas next to one another. Each will allow either rich text and image widgets nested inside.
 
-```js
-// modules/@apostrophecms/home-page/index.js
-//
-// Add our new widget to the "widgets" property for
-// the "main" area
-  fields: {
-    add: {
-      main: {
-        ...
-        options: {
-          widgets: {
-            ...
-            'two-column': {}
-          }
-        }
-      }
-    }
-  }
-```
+First create the module configuration file, extend the core widget type module, and add a widget label for editors. If you do not add a label, Apostrophe will attempt to generate one for the UI based on the module's name.
+
+**The module's name must end in `-widget`.** It is a convention that supports core business logic around widgets and can help keep project code organized. This two-column widget is named `two-column-widget`.
 
 ```js
 // modules/two-column-widget/index.js
 module.exports = {
   extend: '@apostrophecms/widget-type',
   options: {
-    label: 'Two Column'
+    label: 'Two column'
   },
+  // 👇 The widget type's field schema
   fields: {
     add: {
+      // 👇 The first column area
       columnOne: {
         type: 'area',
         label: 'Column One',
         options: {
-          // You can copy the `widgets` option from the `main` area in
-          // home-page/index.js
+          widgets: {
+            '@apostrophecms/rich-text': {},
+            '@apostrophecms/image': {}
+          }
         }
       },
+      // 👇 The second column area
       columnTwo: {
         type: 'area',
         label: 'Column Two',
         options: {
-          // You can copy the `widgets` option from the `main` area in
-          // home-page/index.js
+          widgets: {
+            '@apostrophecms/rich-text': {},
+            '@apostrophecms/image': {}
+          }
         }
       }
     }
   }
 };
+
+```
+
+You may notice there is no `group` property to the field schema. The widget editing interface does not have field groups, so it is not necessary here.
+
+You can then add this module to the `app.js` file to instantiate it.
+
+```js
+// app.js
+require('apostrophe')({
+  shortName: 'my-website',
+  modules: {
+    'two-column-widget': {}
+  }
+});
 ```
 
+## Widget templates
+
+Before using the new widget type, it needs a template file, `widget.html`, in the module's `views` directory. A simple template for the two column widget might look like:
+
 ```django
```

Comment:
should we keep Django here? is that still the most accurate?

Reply:
Yep.